### PR TITLE
Bump fuel-types to v0.3

### PR DESCRIPTION
**Pull request by adlerjohn**
Fix by changing `fuel_types::Color` to `fuel_types::AssetId`, and percolate the changes upwards to the public API of `fuel-tx`.

Depends on
- [x] https://github.com/FuelLabs/fuel-crypto/pull/13
- [x] https://github.com/FuelLabs/fuel-crypto/pull/11
- [x] https://github.com/FuelLabs/fuel-asm/pull/45

### CODE REVIEW

**Comment by Voxelot** [comment]
The public api is already being changed by using this new type, why not change the field names as well?

**Review by vlopes11 — DISMISSED** [review]
One unrelated comment and one requested change

File: tests/bytes.rs [hunk 1]
```diff
@@ -273,6 +273,7 @@ fn receipt() {
         ),
         Receipt::script_result(
             InstructionResult::error(
+                // TODO rename to AssetIdNotFound
```
[the remark this answers is not in the document]
Comment (adlerjohn) [reply tests/bytes.rs]: Fixed in [4857272](https://github.com/FuelLabs/fuel-tx/pull/85/commits/485727282f61a5fb8f4d77b71c7c879dc6da1be1)

File: src/transaction/internals.rs [hunk 2]
```diff
@@ -49,7 +49,10 @@ impl Transaction {
 
     pub(crate) fn _set_script(&mut self, _script: Vec<u8>) -> Option<()> {
         match self {
-            Self::Script { script, .. } => Some(*script = _script),
+            Self::Script { script, .. } => {
```
[the remark this answers is not in the document]
Comment (adlerjohn) [reply src/transaction/internals.rs:52]: That was requested by Clippy, fixed here: [bb93aa0](https://github.com/FuelLabs/fuel-tx/pull/85/commits/bb93aa07fb7d8556212b772852e69137b46fc136)

I'm not actually sure if the change I made is correct, can you double-check?

[the remark this answers is not in the document]
Comment (Voxelot) [reply src/transaction/internals.rs:52]: I think the issue is that this whole method, `_set_script`, and it's params, uses `_` as if it was unused.

[the remark this answers is not in the document]
Comment (adlerjohn) [reply src/transaction/internals.rs:52]: Oh I see. Well so long as this change is fine, we can fix the use of `_` prefix in a future PR.

**Review by Voxelot — APPROVED** [review]

**Pull request merged**